fontir/src/ir.rs

```diff
@@ -28,26 +28,28 @@ pub struct StaticMetadata {
     /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/head>.
     pub units_per_em: u16,
 
-    pub ascender: OrderedFloat<f32>,
-    pub descender: OrderedFloat<f32>,
-    pub cap_height: OrderedFloat<f32>,
-    pub x_height: OrderedFloat<f32>,
-
     /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/name>.
     pub names: HashMap<NameKey, String>,
 
     /// Every axis used by the font being compiled
     pub axes: Vec<Axis>,
+
     /// The name of every glyph, in the order it will be emitted
     ///
     /// <https://rsheeter.github.io/font101/#glyph-ids-and-the-cmap-table>
     pub glyph_order: IndexSet<GlyphName>,
+
     /// A model of how the space defined by [Self::axes] is split into regions that have deltas.
     ///
     /// This copy includes all locations used in the entire font. Users, such as glyph BE, may wish
     /// to narrow (submodel in FontTools terms) to the set of locations they actually use. Use of a
     /// location not in the global model is an error.
     pub variation_model: VariationModel,
+
+    /// The values of metrics at various positions in design space.
+    ///
+    /// At a minimum should be defined at the default location.
+    pub metrics: HashMap<NormalizedLocation, Metrics>,
```

Comment:
I think the reason we called this metadata of IR "_Static_ Metadata" is because it is not variable, but static. At least that's how I understood it. If we now introduce a map of metrics at various locations, it is no longer static, in my view. 
Maybe it should just be called FontMetadata (as opposed to Glyph data) or simply Metadata.

Alternatively we could also think about splitting this off and create a separate object/file called Metrics (which would roughly correspond to Glyphs.app's [GSFontMaster](https://docu.glyphsapp.com/#gsfontmaster)) and is going to be used for things like MVAR (font-wide metrics variations table).

Reply:
I can see already that my previous argument about StaticMetadata being about global font-wide non-variable stuff doesn't hold very well, because we already have a VariationModel in it, which is the cornerstone of variations...

Reply:
actually, the VariationModel itself does not vary in the same sense of the other pieces of data, it is the thing that link them together, so in that sense it makes sense to be in StaticMetadata.

On the other hand, the very docstring at the top of StaticMetadata says:

> /// Global font info that cannot vary.
> ///
> /// For example, upem, axis definitions, etc, as distinct from
> /// metadata that varies across design space such as ascender/descender.

... which precisely contradicts this PR where "metadata that varies across design space such as ascender/descender" are being added to this StaticMetadata struct.

Reply:
I confess I entirely forgot we put that restriction on static metadata. I think I'd be up for metrics being a separate job and IR entry.

If I made that change would this seem OK?

Reply:
> metrics being a separate job and IR entry.

yeah, sounds good

Reply:
Done